doozerlib/distgit.py

```diff
@@ -584,8 +584,19 @@ def push_image(self, tag_list, push_to_defaults, additional_registries=[], versi
 
                     with io.open(push_config, 'w', encoding="utf-8") as pc:
                         pc.write('\n'.join(all_push_urls))
-
-                    mirror_cmd = 'oc image mirror --filter-by-os=amd64 {} {} --filename={}'.format(dr, insecure, push_config)
+                    mirror_cmd = 'oc image mirror '
+                    if filter_by_os is not None:
+                        mirror_cmd += "--filter-by-os={}".format(filter_by_os)
+                    mirror_cmd += " {} {} --filename={}".format(dr, insecure, push_config)
+                    registry_config_file = ''
+                    if registry_config is not None:
+                        for f in os.listdir(registry_config):
+                            if f.endswith(".json"):
+                                registry_config_file = os.path.join(registry_config, f)
+                                break
+                        if registry_config_file == '':
+                            raise FileNotFoundError("Can not find the registry config file in {}".format(registry_config))
```

Comment:
whatever code you end up using, please extract it to a method (perhaps in a util lib) instead of repeating it